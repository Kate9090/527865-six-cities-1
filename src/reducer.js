
const initialState = {
  city: `Amsterdam`,
  offerInCity: require(`./mocks/offers-in-amsterdam`).offerHotelList,
};


const ActionCreator = ({
  'NEW_CITY': (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),

  // 'RESET': () => ({
  //   type: `RESET`,
  // }),

});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return Object.assign({}, state, {
        city: action.payload,
        offerInCity: require(`./mocks/offers-in-${state.city}`).offer,
      });
    // case `RESET`:
    //   return Object.assign({}, state, initialState);
  }

  return state;
};

export {reducer, ActionCreator};
