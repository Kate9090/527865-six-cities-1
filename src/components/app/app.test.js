import React from "react";
import renderer from 'react-test-renderer';

import App from './app.jsx';

import {Provider} from 'react-redux';
import reducer from '../../reducer/index';
import {createStore} from 'redux';

const store = createStore(
    reducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

it(`renders correctly app screen`, () => {
  const tree = renderer
    .create(<Provider store={store}><App
    /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
