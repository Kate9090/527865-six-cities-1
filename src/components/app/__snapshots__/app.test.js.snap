// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly app screen 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="cities tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      />
    </section>
  </div>
  <div
    className="cities__places-wrapper"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          312 places to stay in 
          Cologne
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            className="places__sorting-type"
            tabIndex="0"
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom places__options--opened"
          >
            <li
              className="places__option places__option--active"
              tabIndex="0"
            >
              Popular
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: low to high
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: high to low
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Top rated first
            </li>
          </ul>
          <select
            className="places__sorting-type"
            defaultValue="popular"
            id="places-sorting"
          >
            <option
              className="places__option"
              value="popular"
            >
              Popular
            </option>
            <option
              className="places__option"
              value="to-high"
            >
              Price: low to high
            </option>
            <option
              className="places__option"
              value="to-low"
            >
              Price: high to low
            </option>
            <option
              className="places__option"
              value="top-rated"
            >
              Top rated first
            </option>
          </select>
        </form>
        <div
          className="cities__places-list places__list tabs__content"
        />
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "800px",
              }
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;
